x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/pattern/index_check_flyout/historical_results/historical_results_list/historical_result/index.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiSpacer } from '@elastic/eui';
+
+import type { HistoricalResult as HistoricalResultType } from '../../../../../../../types';
+import { IndexStatsPanel } from '../../../index_stats_panel';
+import { HistoricalCheckFields } from './historical_check_fields';
+// eslint-disable-next-line no-restricted-imports
+import { isNonLegacyHistoricalResult } from './utils/is_non_legacy_historical_result';
+// eslint-disable-next-line no-restricted-imports
+import { LegacyHistoricalCheckFields } from './legacy_historical_check_fields';
+
+export interface Props {
+  indexName: string;
+  historicalResult: HistoricalResultType;
+}
+
+const HistoricalResultComponent: React.FC<Props> = ({ indexName, historicalResult }) => {
+  const {
+    docsCount,
+    sizeInBytes,
+    ilmPhase,
+    ecsFieldCount,
+    totalFieldCount,
+    customFieldCount,
+    checkedAt,
+  } = historicalResult;
+
+  return (
+    <div data-test-subj={`historicalResult-${checkedAt}`}>
+      <EuiSpacer />
+      <IndexStatsPanel
+        docsCount={docsCount}
+        sizeInBytes={sizeInBytes ?? 0}
+        ilmPhase={ilmPhase}
+        ecsCompliantFieldsCount={ecsFieldCount}
+        customFieldsCount={customFieldCount}
+        allFieldsCount={totalFieldCount}
+      />
+      <EuiSpacer />
+      {isNonLegacyHistoricalResult(historicalResult) ? (
```

Comment:
you are absolutely right, and that's what I do everywhere and push for everywhere so thanks for that.

Having said that I had a reason :D

Unfortunately there is a problem with TS compilers abiility to type guard from negated legacy check

![image](https://github.com/user-attachments/assets/647624b7-4c43-4c98-ad3a-d88908d23f46)

It cannot yield the correct type from here because of the negation.

But if I remove the negation it works, and if I do that I have to use the NonLegacyHistoricalResult.

Now you know the context.

If this is critical I can rename the NonLegacy to just HistoricalResult but that will the chain of types everywhere, and I will to come up with a replacement to existing HistoricalResult type which encompasses both Historicalresult and LegacyHistoricalResult

Reply:
I would suggest we keep it as it for now and revise later

Reply:
What about this?
```
return (
    !('incompatibleFieldMappingItems' in historicalResult) ||
    !('incompatibleFieldValueItems' in historicalResult) ||
    !('sameFamilyFieldItems' in historicalResult)
  );
```

Reply:
nope I tried.

Reply:
What is the TS issue? It works for me

Reply:
![image](https://github.com/user-attachments/assets/be7ffba1-2ab8-4d04-9b1e-3203b37b2172)
![image](https://github.com/user-attachments/assets/29d7caa0-ec65-40b0-9e5d-98940b7364bc)

Yes but accidentally, because it's a type booby-trap as seen in the screenshots above :) 

So next person that decides for example to refactor this and change the interface of `<HistoricalCheckFields />` and pass let's say incompatibleFieldMappingItems directly into props. Will suddenly discover that the types are inferred wrong.

![image](https://github.com/user-attachments/assets/39a53a8a-5949-428d-93bc-d83f06d1f05b)

And the reason it doesn't error is because typescript :) namely:

1) In first branch of isLegacyCheck it returns HistoricalResult, which is wider than LegacyHistoricalResult, but minimally structurally compatible with LegacyHistoricalResult type, because TS does structure based validation, so it doesn't error even tho it's mistyped.

2) Else branch is `never` and since `never` is assignable to anything it also is not throwing any type error.

So on paper it works, but on accident and it's superfragile so I'd rather not plant this dirty bomb for somebody else.

I hope this clarifies.